File: 2024/April/4-15 129 Sum Root To Leaf Numbers/SumToLeaf.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return recursiveHelper(root, "");
    }

    private int recursiveHelper(TreeNode node, String val) {
        if(node == null) {
            return 0;
        }

        val += node.val;

        if(node.left == null && node.right == null) {
            return Integer.valueOf(val);
        }

        return recursiveHelper(node.left, val) + recursiveHelper(node.right, val);
    }
}

File: 2024/April/4-15 129 Sum Root To Leaf Numbers/SumToLeaf.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumNumbers = function(root) {
    var recursiveHelper = (node, val) => {
        if(!node) {
            return 0;
        }

        val += node.val;

        if(!node.left && !node.right) {
            return parseInt(val);
        }

        return recursiveHelper(node.left, val) + recursiveHelper(node.right, val);
    };

    return recursiveHelper(root, "");
};